src/client/InteractionClient.js

```diff
@@ -0,0 +1,241 @@
+   * @param {Snowflake} [guildID] Optional guild ID.
+   * @returns {ApplicationCommand[]}
+   */
+  async fetchCommands(guildID) {
+    let path = this.client.api.applications(this.applicationID);
+    if (guildID) {
+      path = path.guilds(guildID);
+    }
+    const commands = await path.commands.get();
+    return commands.map(c => new ApplicationCommand(this, c, guildID));
+  }
+
+  /**
+   * Options for a slash command.
+   * @typedef {Object} ApplicationCommandOptions
+   * @property {string} type The type of the slash command
+   * @property {string} name The name of the slash command
+   * @property {string} description The description of the slash command
+   * @property {boolean} [required=false] Whether or not the option is required ot optional
+   * @property {ApplicationCommandOptionChoice[]} [choices] The choices of the slash command for the user to pick from
+   * @property {ApplicationCommandOptions[]} [options] Additional options if the slash command is a subcommand or a
+   * subcommand group
+   */
+
+  /**
+   * Choices for a slash command option
+   * @typedef {Object} ApplicationCommandOptionChoice
+   * @property {string} name The name of the choice
+   * @property {string|number} value The value of the choice
+   */
+
+  /**
+   * Set all the commands for the application or guild.
+   * @param {ApplicationCommandOptions[]} commands The command descriptor.
+   * @param {Snowflake} [guildID] Optional guild ID.
+   * @returns {ApplicationCommand[]} The commands.
+   */
+  async setCommands(commands, guildID) {
+    let path = this.client.api.applications(this.applicationID);
+    if (guildID) {
+      path = path.guilds(guildID);
+    }
+    const cs = await path.commands.put({
+      data: commands.map(transformCommand),
+    });
+    return cs.map(c => new ApplicationCommand(this, c, guildID));
+  }
+
+  /**
+   * Create a command.
+   * @param {ApplicationCommandOptions} command The command descriptor.
+   * @param {Snowflake} [guildID] Optional guild ID.
+   * @returns {ApplicationCommand} The created command.
+   */
+  async createCommand(command, guildID) {
+    let path = this.client.api.applications(this.clientID);
+    if (guildID) {
+      path = path.guilds(guildID);
+    }
+    const c = await path.commands.post({
+      data: transformCommand(command),
+    });
+    return new ApplicationCommand(this, c, guildID);
+  }
+
+  async handle(data) {
+    switch (data.type) {
+      case InteractionType.PING:
+        return {
+          type: InteractionResponseType.PONG,
+        };
+      case InteractionType.APPLICATION_COMMAND: {
+        let timedOut = false;
+        let resolve;
+        const directPromise = new Promise(r => {
+          resolve = r;
+          this.client.setTimeout(() => {
+            timedOut = true;
+            r({
+              type: InteractionResponseType.DEFERRED_CHANNEL_MESSAGE_WITH_SOURCE,
+            });
+          }, 250);
+        });
+
+        const syncHandle = {
```

Comment:
why not just use the reply method directly here?

`const syncReply = resolved => ...;`

Reply:
it had other methods originally. might again, not a worthwhile refactor imo.

Reply:
> why not just use the reply method directly here?
> 
> 
> 
> `const syncReply = resolved => ...;`